File: API/Models/Roles.cs
/* File: Roles.cs
 * Purpose: Single source of truth for the 3 exact role string values used across the system.
 * Author: IT23218512 
 */

namespace API.Models
{
    public static class Roles
    {
        public const string Backoffice = "Backoffice";
        public const string GridOperator = "GridOperator";
        public const string Prosumer = "Prosumer";
    }
}

File: Mobile/app/src/main/java/com/example/smart_solar_mobile/models/Roles.java
// File: Roles.java
// Purpose: Role names exactly as the API sends them in the login response.
// Author: IT23215856

package com.example.smart_solar_mobile.models;

public final class Roles {
    public static final String PROSUMER = "Prosumer";
    public static final String GRID_OPERATOR = "GridOperator";
    public static final String BACKOFFICE = "Backoffice";

    private Roles() {
        // Constants only, never instantiated
    }
}
